File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/stock/transaction_operation/stock_in/TOStockMovementStockInResDTO.java
package com.example.fieldsync_inventory_api.dto.stock.transaction_operation.stock_in;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

// Transaction Operation Stock Movement Stock Input Response DTO
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TOStockMovementStockInResDTO {
    private Long id;
    private String movementType;
    private Long seedBatchId; // For create new if not found
    private BigDecimal purchaseItem; // stockMovementId and saleId will be ignored

    private BigDecimal quantity;
    private String description;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/transaction_operation/response/stock_in/TOStockMovementStockInResDTO.kt
package com.example.fieldsync_inventory_app.data.remote.dto.transaction_operation.response.stock_in

import java.math.BigDecimal

data class TOStockMovementStockInResDTO(
    val id: Long,
    val movementType: String,
    val seedBatchId: Long, // For create new if not found
    val purchaseItem: BigDecimal, // stockMovementId and purchaseId will be ignored
    val quantity: Double,
    val description: String
)
